pittapi/course_helper.py

```diff
@@ -0,0 +1,54 @@
+from typing import List
+import re
+import requests
+
+SUBJECT_CODES_API = "https://prd.ps.pitt.edu/psc/pitcsprd/EMPLOYEE/SA/s/WEBLIB_HCX_CM.H_COURSE_CATALOG.FieldFormula.IScript_CatalogSubjects?institution=UPITT"
+SUBJECT_COURSES_API = "https://prd.ps.pitt.edu/psc/pitcsprd/EMPLOYEE/SA/s/WEBLIB_HCX_CM.H_COURSE_CATALOG.FieldFormula.IScript_SubjectCourses?institution=UPITT&subject={subject}"
+COURSE_DETAIL_API = "https://prd.ps.pitt.edu/psc/pitcsprd/EMPLOYEE/SA/s/WEBLIB_HCX_CM.H_COURSE_CATALOG.FieldFormula.IScript_CatalogCourseDetails?institution=UPITT&course_id={id}&effdt=2018-06-30&x_acad_career={career}&crse_offer_nbr=1&use_catalog_print=Y"
+COURSE_SECTIONS_API = "https://prd.ps.pitt.edu/psc/pitcsprd/EMPLOYEE/SA/s/WEBLIB_HCX_CM.H_BROWSE_CLASSES.FieldFormula.IScript_BrowseSections?institution=UPITT&campus=&location=&course_id={id}&institution=UPITT&x_acad_career={career}&term={term}&crse_offer_nbr=1"
+    # id -> unique course ID, not to be confused with course code (for instance, CS 0007 has code 105611)
+    # career -> for example, UGRD (undergraduate)
+
+
+def _validate_term(term: str) -> str:
+    """Validates that the term entered follows the pattern that Pitt does for term codes."""
+    valid_terms = re.compile("2\d\d[147]")
```

Comment:
Don't compile the regex each time this function is called. Compile it once during module import at the top and then just use it.

Reply:
.NET 7 has a regex-generator that does this automatically at compile time. Is there a similar thing available for Python?